cli-commands/compiler/vyperCompiler/vyperCompiler.js

```diff
@@ -0,0 +1,97 @@
+const dockerCLI = require('docker-cli-js');
+const docker = new dockerCLI.Docker();
+const path = require("path");
+const fs = require("fs");
+const mkdirp = require("mkdirp");
+const find_contracts = require("./../etherlime-contract-sources");
+
+
+const run = async (allFiles, buildDirectory) => {
+
+    for (let i = 0; i < allFiles.length; i++) {
+        try {
+            let filePath = path.normalize(path.relative(process.cwd(), allFiles[i])) //extract pure file path
+            let fileBaseName = path.basename(filePath, ".vy")
+            let fileTimestampStatus = await getFileTimestampStatus(filePath)
+
+            if (!await isFileUpdated(fileBaseName, fileTimestampStatus, buildDirectory)) {
+                return
+            }
+
+            let compiledObject = await compile(filePath)
+            compiledObject = Object.assign({"contractName": fileBaseName}, compiledObject[filePath]) //restructuring the compiled object
+            compiledObject.updatedAt = fileTimestampStatus
+
+            await recordCompiledObject(compiledObject, buildDirectory)
+
+            const displayPath = "." + path.sep + filePath;
+            console.log(`Compiling ${displayPath}...`);
+        } catch (e) {
+            throw new Error("Vyper compilation failed." + e)
+        }
+    }
+
+}
+
+//gets timestamp indicating the last time the file was changed or modified
+const getFileTimestampStatus = async (filePath) => {
+    try {
+        let stats = fs.statSync(filePath)
+        return (stats.ctime || stats.mtime).getTime()
```

Comment:
Wont this always give you the creation time instead of the modification time?

Reply:
ctime is not creation time but change time and mtime is modification time. In most cases both are equal. mtime is updated only when file's content is changed, and ctime - also when we change file’s permissions, name or location